### PR TITLE
Ensure that all test suites work on stable

### PR DESCRIPTION
**Pull request by Xarthisius**
This PR should be merged before #2800 It ports all the necessary changes to testing infra in order to ensure that current stable branch passes all tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -95,6 +87,9 @@ jobs:
       python: 3.8
       script: coverage run $(which nosetests) -c nose_unit.cfg
 
+      # This is not necessarily going to be forever -- once we merge yt-4.0
```
Comment (matthewturk) [inline .travis.yml:90]: hmmm this seems out of date?

**Review by matthewturk — APPROVED** [review]
seems good to me

File: .travis.yml (hunk 1 above)
Comment (Xarthisius) [reply .travis.yml:90]: That was copied from current master :) For the sake of future merge conflicts I'd leave it without any changes

**Review by munkm — APPROVED** [review]
Awesome. Thank you so much for doing this Kacper!

**Pull request merged**